kernel/src/transaction/mod.rs

```diff
@@ -27,10 +37,26 @@ pub(crate) static ADD_FILES_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
         StructField::not_null("size", DataType::LONG),
         StructField::not_null("modificationTime", DataType::LONG),
         StructField::not_null("dataChange", DataType::BOOLEAN),
-        StructField::nullable("numRecords", DataType::LONG),
     ]))
 });
 
+/// Returns a reference to the mandatory fields in an add action.
+pub(crate) fn mandatory_add_file_schema() -> &'static SchemaRef {
+    &MANDATORY_ADD_FILE_SCHEMA
+}
+
+/// The static instance referenced by [`add_files_schema`].
+pub(crate) static ADD_FILES_SCHEMA: LazyLock<SchemaRef> = LazyLock::new(|| {
```

Comment:
Note that I kept the name `ADD_FILES_SCHEMA` here since we discussed this in another PR.
I still think the naming is not super clear, but I would rather address this in a separate PR.